src/science/canopy/cbl_qsat.F90

```diff
@@ -1,19 +1,37 @@
 MODULE cbl_qsat_module
+  !* This MODULE contains two SUBROUTINEs that calculate
+  ! the specific humdity at saturation as a function of air pressure
+  ! and temperature.
+  !
+  ! The two SUBROUTINEs differ only in that [[qsatfjh]] operates on an array
+  ! whereas [[qsatjfh2]] operates on single element REALs.
 
 PUBLIC qsatfjh
 PUBLIC qsatfjh2
 
 CONTAINS
 
-SUBROUTINE qsatfjh(mp, var, CRMH2o, Crmair, CTETENA, CTETENB, CTETENC, tair,pmb)
+  SUBROUTINE qsatfjh(mp, var, CRMH2o, Crmair, CTETENA, CTETENB, CTETENC, tair,pmb)
+    !*## Purpose
+    !
+    ! This SUBROUTINE evaluates the specific humdity (water vapour mixing ratio)
+    ! at saturation (in kgkg\(^{-1}\)), at a given air temperature (in \(^{\circ}\)C)
+    ! and air pressure (in hPa) for an array of size mp.
+    !
+    !## Method
+    !
+    ! The standard formula for specific humdity at saturation is used,
+    ! see e.g. Garratt (1995), based on the values for the mass of a mole
+    ! of water,`CRMH2o`, mass of a mole of dry air `Crmair`,
+    ! and the Teten constants.
```

Comment:
It took me a while to follow the citation (still haven't found it I think). What about replacing with:

```suggestion
    ! The Teten's formula for specific humdity at saturation is used,
    ! based on the values for the mass of a mole of water,`CRMH2o`, 
    ! mass of a mole of dry air `Crmair`, and the Teten constants.
    !
    !## Reference
    !
    ! [Murray F.W, 1967](https://doi.org/10.1175/1520-0450(1967)006%3C0203:OTCOSV%3E2.0.CO;2)
```
I found the reference on Wikipedia but checked it. And it's referred to in the constants file:

> ! Magnus Tetans (Murray 1967)

(No comment on the typo in the comment in `cable_phys_constants_mod.F90`